flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobUtils.java

```diff
@@ -227,7 +227,7 @@ static File getStorageLocation(File storageDir, JobID jobID, String key) {
 	private static File getJobDirectory(File storageDir, JobID jobID) {
 		final File jobDirectory = new File(storageDir, JOB_DIR_PREFIX + jobID.toString());
 
-		if (!jobDirectory.exists() && !jobDirectory.mkdirs()) {
```

Comment:
Why did you change the order of those operations? Is that fix for something? If so could you add explanation to the commit message what's going on here?